controllers/machine_controller.go

```diff
@@ -183,9 +185,15 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	} else {
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
-		if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
-			klog.Errorf("Error deleting node %q for machine %q: %v", m.Status.NodeRef.Name, m.Name, err)
-			return ctrl.Result{}, err
+		err = wait.PollImmediate(time.Second*2, time.Second*10, func() (bool, error) {
```

Comment:
```suggestion
		err = wait.PollImmediate(2*time.Second, 10*time.Second, func() (bool, error) {
```

Super nit, for readability, it'll read as "2 seconds" and "10 seconds"